R/core_arrow.R

```diff
@@ -7,47 +9,79 @@ arrow_write_record_batch <- function(df) {
     record <- record_batch(df)
     write_record_batch(record, raw())
   }
-  if (packageVersion("arrow") <= "0.13") {
+  else if (packageVersion("arrow") <= "0.13") {
     record <- record_batch(df)
 
     write_arrow <- get("write_arrow", envir = as.environment(asNamespace("arrow")))
     write_arrow(record, raw())
   }
   else {
-    record <- record_batch(!!!df)
+    environment <- list()
+    if (packageVersion("arrow") >= "0.15") {
```

Comment:
This won't work because the "devel" version is 0.14.1.9000. I'd just set this unconditionally, it won't do anything for older versions.